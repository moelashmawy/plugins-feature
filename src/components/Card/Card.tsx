import classNames from 'classnames';
import { Switch } from '../Switch';
import { Loading } from '../Loading/Loading';

type Props = {
  isDisabled: boolean;
  title: string;
  isChecked: boolean;
  switchDisabled: boolean;
  isLoading: boolean;
  status: string;
  description: string;
  onSwitch: () => void;
};

export const Card = ({
  isDisabled,
  title,
  isChecked,
  switchDisabled,
  isLoading,
  status,
  description,
  onSwitch,
}: Props) => {
  return (
    <div
      className={classNames('card-component', { 'disabled-card': isDisabled })}
    >
      <div className="header">
        <h3>{title}</h3>
        <Switch
          isChecked={isChecked}
          onChange={onSwitch}
          disabled={switchDisabled}
        />
      </div>

      <div className={classNames('card-status', { inactive: !isChecked })}>
        <span className="loading">{isLoading && <Loading />}</span>
        <span>{status}</span>
      </div>

      <div className="desc">
        <p>{description}</p>
      </div>
    </div>
  );
};
