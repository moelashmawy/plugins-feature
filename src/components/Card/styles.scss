.card-component {
  border: 3px solid #ddd;
  border-radius: 10px;
  padding: 18px;
  height: 120px;
  min-height: 170px;

  .header {
    display: flex;
    justify-content: space-between;
    align-items: center;

    h3 {
      font-weight: 500;
    }
  }

  .card-status {
    color: $color-allowed;
    margin-top: -10px;
    display: flex;
    justify-content: end;

    &.inactive {
      color: $color-blocked;
    }

    .loading {
      margin-right: 5px;
    }
  }

  .desc {
    display: flex;
    margin-top: -10px;

    p {
      flex: 0 0 85%;
    }
  }
}

.disabled-card {
  opacity: 50%;
  cursor: not-allowed;
}
