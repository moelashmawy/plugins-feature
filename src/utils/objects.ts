export const getKeyByPropName = (
  data: any,
  keyToFind: string,
  prop: string
): string => {
  const tabName = Object.keys(data).find(
    (tabName) => data[tabName][prop] === keyToFind
  )!;
  return tabName;
};
