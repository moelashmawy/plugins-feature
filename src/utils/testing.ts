import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { expect } from 'vitest'
import { configureStore } from '@reduxjs/toolkit';
import pluginsSLice from '../state/slices/pluginsSLice';

export const createSnapshot = (ComponentToTest: ReactElement) => {
    const { asFragment } = render(ComponentToTest);
    expect(asFragment()).toMatchSnapshot();
}

export const createTestStore = () => {
  const store = configureStore(
    {
      reducer: {
        plugins: pluginsSLice
      }
    }
  );
  return store;
}