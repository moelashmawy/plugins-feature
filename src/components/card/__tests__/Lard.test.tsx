import { it } from 'vitest'
import { createSnapshot } from '../../../utils/testing';
import { Card } from '../Card';

it('should render plugin card correctly', () => {
  createSnapshot(
    <Card
        title='plugin 1'
        description='Awesome plugin'
        status={"Allowed"}
        isDisabled={false}
        isChecked={true}
        isLoading={false}
        switchDisabled={false}
        onSwitch={() => {}}
    />
  );
});