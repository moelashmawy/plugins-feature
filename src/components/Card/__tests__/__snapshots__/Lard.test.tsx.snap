// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render plugin card correctly 1`] = `
<DocumentFragment>
  <div
    class="card-component"
  >
    <div
      class="header"
    >
      <h3>
        plugin 1
      </h3>
      <label
        class="switch"
      >
        <input
          checked=""
          type="checkbox"
        />
        <span
          class="slider round"
        />
      </label>
    </div>
    <div
      class="card-status"
    >
      <span
        class="loading"
      />
      <span>
        Allowed
      </span>
    </div>
    <div
      class="desc"
    >
      <p>
        Awesome plugin
      </p>
    </div>
  </div>
</DocumentFragment>
`;
